comp/core/profiler/impl/profiler_mock.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build test
+
+package profilerimpl
+
+import (
+	"net/http"
+)
+
+// NewMockHandler provides a standard handler that will mimic the endpoints the profiler will scan for pprof data
+// Can be injected directly in to an httptest server
+func NewMockHandler() http.HandlerFunc {
```

Comment:
This should be in a `mock` folder so people can depend on the mock without pulling the implementation (ref https://datadoghq.dev/datadog-agent/components/creating-components/#the-mock-folder).